firefox-ios/firefox-ios-tests/Tests/XCUITests/FakespotTests.swift

```diff
@@ -3,35 +3,94 @@
+                           "How reliable are these reviews?")
+            XCTAssertTrue(app.staticTexts["Adjusted rating"].exists)
+            validateHighlightsSection()
         } else {
-            navigator.openURL("https://www.amazon.com")
-            waitUntilPageLoad()
-
-            // Search for and open a shoe listing
-            let website = app.webViews["contentView"].firstMatch
-            mozWaitForElementToExist(website.textFields["Search Amazon"])
-            XCTAssert(website.textFields["Search Amazon"].isEnabled)
-            website.textFields["Search Amazon"].tap()
-            website.textFields["Search Amazon"].typeText("Shoe")
-            website.buttons["Go"].tap()
-            waitUntilPageLoad()
-            website.images.firstMatch.tap()
-
-            // Tap the shopping cart icon
-            waitUntilPageLoad()
-            mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.shoppingButton])
-            app.buttons[AccessibilityIdentifiers.Toolbar.shoppingButton].tap()
-            mozWaitForElementToExist(app.staticTexts[AccessibilityIdentifiers.Shopping.sheetHeaderTitle])
-            XCTAssertEqual(app.staticTexts[AccessibilityIdentifiers.Shopping.sheetHeaderTitle].label, "Review Checker")
-
-            // Close the popover
-            app.otherElements.buttons[AccessibilityIdentifiers.Shopping.sheetCloseButton].tap()
-            mozWaitForElementToNotExist(app.otherElements[AccessibilityIdentifiers.Shopping.sheetHeaderTitle])
+            XCTAssertTrue(app.staticTexts["No info about these reviews yet"].exists)
+            XCTAssertTrue(app.buttons["Check Review Quality"].exists)
         }
+        mozWaitForElementToExist(app.staticTexts[AccessibilityIdentifiers.Shopping.ReviewQualityCard.title])
+        XCTAssertEqual(app.staticTexts[AccessibilityIdentifiers.Shopping.ReviewQualityCard.title].label, 
+                       "How we determine review quality")
+        XCTAssertEqual(app.staticTexts[AccessibilityIdentifiers.Shopping.SettingsCard.title].label, "Settings")
+        XCTAssertTrue(app.staticTexts["Review Checker is powered by Fakespot by Mozilla"].exists)
+    }
+
+    private func validateHighlightsSection() {
+        if app.staticTexts[AccessibilityIdentifiers.Shopping.HighlightsCard.title].exists {
+            XCTAssertEqual(app.staticTexts[AccessibilityIdentifiers.Shopping.HighlightsCard.title].label, 
+                           "Highlights from recent reviews")
+            if app.staticTexts["Show More"].exists {
+                app.staticTexts["Show More"].tap()
+                if app.staticTexts[AccessibilityIdentifiers.Shopping.HighlightsCard.groupPriceTitle].exists {
+                    XCTAssertEqual(app.staticTexts[AccessibilityIdentifiers.Shopping.HighlightsCard.groupPriceTitle].label, 
+                                   "Price")
+                } else if app.staticTexts[AccessibilityIdentifiers.Shopping.HighlightsCard.groupQualityTitle].exists {
```

Comment:
I am a bit concern about these else if statment if something exists... since we don't wait for elements it may be flaky at some point. Let's monitor this closely in case we have to change this.

Reply:
As you can see locally i had no problems, but indeed it needs a close attention, the behaviour may be different on your CI machines.